src/UpdateHelper/UpdateHelper.php

```diff
@@ -71,42 +71,42 @@ protected static function getUpdateHelperConfig(Composer $composer, $key = null)
 
     public static function check(Event $event)
     {
-        if ($event instanceof ScriptEvent || $event instanceof PackageEvent) {
-            $io = $event->getIO();
-            $composer = $event->getComposer();
-            $autoload = __DIR__.'/../../../../autoload.php';
-
-            if (file_exists($autoload)) {
-                include_once $autoload;
-            }
-
-            $classes = static::getUpdateHelperConfig($composer);
+        if (!($event instanceof ScriptEvent) && !($event instanceof PackageEvent)) {
+            return;
+        }
 
-            foreach ($classes as $file => $class) {
-                $error = null;
+        $io = $event->getIO();
+        $composer = $event->getComposer();
+        $autoload = $composer->getConfig()->get('vendor-dir').'/autoload.php';
 
-                if (is_string($class) && class_exists($class)) {
-                    try {
-                        $helper = new $class();
-                    } catch (Exception $e) {
-                        $error = $e->getMessage()."\nFile: ".$e->getFile()."\nLine:".$e->getLine()."\n\n".$e->getTraceAsString();
-                    } catch (Throwable $e) {
-                        $error = $e->getMessage()."\nFile: ".$e->getFile()."\nLine:".$e->getLine()."\n\n".$e->getTraceAsString();
-                    }
+        if (file_exists($autoload)) {
+            include_once $autoload;
+        }
 
-                    if (!$error && $helper instanceof UpdateHelperInterface) {
-                        $helper->check(new static($event, $io, $composer));
+        $classes = static::getUpdateHelperConfig($composer);
 
-                        continue;
-                    }
-                }
+        foreach ($classes as $file => $class) {
+            if (!is_string($class) || !class_exists($class)) {
+                $io->writeError(JsonFile::encode($class).' is not a class in file '.$file);
+                continue;
+            }
 
-                if (!$error) {
-                    $error = JsonFile::encode($class).' is not an instance of UpdateHelperInterface.';
-                }
+            try {
+                $helper = new $class();
+            } catch (Exception $e) {
+                $io->writeError($e->getMessage()."\nFile: ".$e->getFile()."\nLine:".$e->getLine()."\n\n".$e->getTraceAsString());
+                continue;
+            } catch (Throwable $e) {
+                $io->writeError($e->getMessage()."\nFile: ".$e->getFile()."\nLine:".$e->getLine()."\n\n".$e->getTraceAsString());
+                continue;
+            }
 
-                $io->writeError('UpdateHelper error in '.$file.":\n".$error);
```

Comment:
Here we lost the "UpdateHelper error", as different packages could fail in Composer, a prefix for errors is a good way to identify the package that throws it.

Reply:
I see.

Reply:
Could you make the test pass?

Reply:
I found a way that remove the duplications and keep the error prefix.